Umbraco v8

ConstroUmbraco opened the pull request:
Hello,
We have to need LeBlender package which helps us on our Umbraco version 8. If you have created the package then please provide to us.

Contact Email :- constro.umbraco@gmail.com

Thanks in advance.

mirkomaty committed:
A first throw of porting LeBlender to Umbraco V8

mirkomaty committed:
Some formatting to make the dialogs look like Umbraco v8

mirkomaty committed:
Update README.md

mirkomaty committed:
Fix rendering a grid control in the grid.

mirkomaty committed:
Merge branch 'umbraco-v8' of https://github.com/mirkomaty/LeBlender into umbraco-v8

mirkomaty committed:
Upgrade the code to work with Umbraco 8.0.0 release.

mirkomaty committed:
Building the Nuget package. Note: If you test build.bat, make sure to comment out the line which will publish the package.

mirkomaty committed:
Bug: Composer was missing to register the IComponent implementation.
Rename of UmbracoEvents.cs in ApplicationInitializer.cs.

mirkomaty committed:
Umbraco 8: Checkin missing UmbracoCms.Web package.

mirkomaty committed:
Bug: /App_Plugins/LeBlender/Web.config causes troubles. Removing the file seems to work.
Bug: ViewData unknown: Views using ViewData must explicitly inherit from WebViewPage<dynamic>.

mirkomaty committed:
Merge branch 'umbraco-v8' of https://github.com/mirkomaty/LeBlender into umbraco-v8

mirkomaty committed:
Bug: Replaced notificationsService.success method with .then. Fixed with revert.

mirkomaty committed:
Umbraco 8: Make the backoffice content id available in the views

mirkomaty committed:
Umbraco 8: Make the backoffice content id available in the views

mirkomaty committed:
Umbraco 8.0.1: The custom MVC routes are back again, so our workaround crashed. We first stick to the workaround.

mirkomaty committed:
Umbraco 8.0.1: We needed a new way of creating a PublishedPropertyType to determine the right PropertyValueConverter.

soreng commented:
Hi @ConstroUmbraco 

I have tried out this PR. It builds fine, but I can't get it running. This is mainly because the test-site is still running umbraco v7.3.4. Could you update this, to enable testing of the project?

ConstroUmbraco commented:
Hello,
Can you please suggest a step how can we use this package in Umbraco 8

soreng commented:
Uhm, that was more or less the question i asked you @ConstroUmbraco 

The test-site in the solution can't start as it is in the PR :)

ConstroUmbraco commented:
Hello,
PR stands for?

soreng commented:
> PR stands for?

https://help.github.com/en/articles/about-pull-requests

the thing you opened here :)

ConstroUmbraco commented:
Ohhhh yes got it.
@soreng you have to create an installation package of LeBlender then Please provide to us because it will help us in our Umbraco version 8.

bjarnef commented:
@ConstroUmbraco it seems it has a few issues with icon picker.
![image](https://user-images.githubusercontent.com/2919859/57080759-82462a80-6cf3-11e9-824a-5deb679925a2.png)

I will suggest remove the "Editor icon" property/row and just handle it from heading similar to document types to make it consistent.

![image](https://user-images.githubusercontent.com/2919859/57080904-e537c180-6cf3-11e9-91ac-ec92a671abf6.png)

bjarnef commented:
Furthermore the infinite editor should not use icon picker, but from configuration view instead.

![image](https://user-images.githubusercontent.com/2919859/57101160-e3d0be00-6d20-11e9-8e6f-f9896cbc0f1e.png)

Also consider to re-use layout as in document types for header (name, description, alias and icon) and fixed footer with save button.

![image](https://user-images.githubusercontent.com/2919859/57101570-d0722280-6d21-11e9-9181-87a11407a6d9.png)

bjarnef commented:
I have cleaned up the UI a bit by re-using more of the core UI components and updated the look and feel to v8.

![image](https://user-images.githubusercontent.com/2919859/57108042-f0a9dd80-6d31-11e9-9be0-cdbe9a341cc7.png)

![image](https://user-images.githubusercontent.com/2919859/57108067-facbdc00-6d31-11e9-8247-99d01266151c.png)

![image](https://user-images.githubusercontent.com/2919859/57108093-0fa86f80-6d32-11e9-8cc5-3d1c63252fd0.png)

bjarnef commented:
I will also suggest we fix the typo in `propretyType`.
https://github.com/Lecoati/LeBlender/issues/29

and remove the `legacy adapter`.

```
/***************************************/
        /* legacy adaptor 0.9.15 */
        /***************************************/

		var parameter = $scope.model.dialogData.parameter;

        if (parameter && parameter.propretyType) {

            switch (parameter.propretyType.name) {
                case "Textstring":
                    parameter.dataType = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
                    parameter.propretyType = {};
                    break;
                case "Textarea":
                    parameter.dataType = "c6bac0dd-4ab9-45b1-8e30-e4b619ee5da3";
                    parameter.propretyType = {};
                    break;
                case "Rich Text Editor":
                    parameter.dataType = "ca90c950-0aff-4e72-b976-a30b1ac57dad";
                    parameter.propretyType = {};
                    break;
                case "Boolean":
                    parameter.dataType = "92897bc6-a5f3-4ffe-ae27-f2e7e33dda49";
                    parameter.propretyType = {};
                    break;
                case "Media Picker":
                    parameter.dataType = "93929b9a-93a2-4e2a-b239-d99334440a59";
                    parameter.propretyType = {};
                    break;
                case "Multi Media Picker":
                    parameter.dataType = "7e3962cc-ce20-4ffc-b661-5897a894ba7e";
                    parameter.propretyType = {};
                    break;
                case "Content Picker":
                    parameter.dataType = "a6857c73-d6e9-480c-b6e6-f15f6ad11125";
                    parameter.propretyType = {};
                    break;
                case "Multi Content Picker":
                    parameter.dataType = "";
                    break;
            }

        }
```

bjarnef commented on Src/Lecoati.LeBlender.Extension/Models/LeBlenderPropertyModel.cs:
```diff
@@ -28,36 +30,23 @@ public class LeBlenderPropertyModel
+				// We skip the code and try to convert the Value property directly.
+
+				// Value is not final value type, so try a regular type conversion aswell
+				var convertAttempt = Value.TryConvertTo<T>();
```
This might work with simple types, but not e.g. `.GetValue<IPublishedContent>` (content picker, media picker (single), etc.)

Maybe have a look at the core property value converters https://github.com/umbraco/Umbraco-CMS/tree/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Core/PropertyEditors

Not sure if there is an recommended approach to get value converter supported in Umbraco core, but there are something like this in core:
https://github.com/umbraco/Umbraco-CMS/blob/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Core/PropertyEditors/PropertyValueConverterCollection.cs#L30-L32

@soreng do you have any thoughs on this?

mirkomaty commented:
@bjarnef: I'm at Codegarden. Are you here, too? Perhaps we can meet and talk?

bjarnef commented:
@mirkomaty unfortunately not this year 😥😐🦄

bjarnef commented:
@mirkomaty I modified the UI part some time ago to make it fit Umbraco V8 and cleaned up some old code. Feel free to update your branch with the changes you can use.
Not sure if this PR is being working on otherwise maybe submit a new PR to fix e.g. issues with icons.
[LeBlender.zip](https://github.com/Lecoati/LeBlender/files/3231158/LeBlender.zip)

mirkomaty commented:
Hi Bjarne, thanks a lot for your contribution. As soon as I find some time I'll incorporate the changes in my work and test it.

bjarnef commented:
@mirkomaty you could also open a "draft" PR with your latest changes at that time, which is open for collaboration and then change it to a "open" PR, when you feel it is ready.

mirkomaty committed:
Added an error message to the DataTypeController

mirkomaty commented:
@bjarnef: I merged your changes in my testproject, but unfortunately the code doesn't work. After saving I don't get the name, the alias, and the icon into the grid.editors.config.js. It seems that there is something missing in the controller (edit.js). A new grid editor using your code looks like that:
```
{
	"name": "",  // <--- missing
	"alias": "",    // <--- missing
	"view": "/App_Plugins/LeBlender/editors/leblendereditor/LeBlendereditor.html",
	"icon": "icon-settings-alt",  // <--- default value instead of my selected icon
	"render": "/App_Plugins/LeBlender/editors/leblendereditor/views/Base.cshtml",
	"config": {
		"frontView": "MyNewEditorView"
	}
}
```
Existing editors don't show their icon and name in the UI.

Could you check, if your LeBlender.zip contains your newest changes?

bjarnef commented:
@mirkomaty yes, it needs to update these default properties.

```
vm.content = {
      name: "",
      alias: "",
      description: "",
      icon: "icon-settings-alt"
};
```

```
// Init model value
// Init model value
$scope.model = {
	value: {
		name: "",
		alias: "",
		view: "",
		icon: "icon-settings-alt"
	}
};

// init autoPopulateAlias
$scope.initAutoPopulateAlias = function () {
	if ($scope.model.value.name === "" && $scope.model.value.name === "") {
		$scope.$watch("model.value.name", function () {
			$scope.model.value.alias = $scope.autoPopulateAlias($scope.model.value.name);
		})
	}
}
```

It needs to refactor the code a bit and clean up stuff, e.g. we don't need `autoPopulateAlias` and the functions `toCamelCase()` and `toPascalCase`, when re-using the `umb-editor-header`.

I will recommend to have a look at the edit view + controller on document types in Umbraco core. Much of the code in LeBlender can be simplified.

https://github.com/umbraco/Umbraco-CMS/blob/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Web.UI.Client/src/views/documenttypes/edit.html

https://github.com/umbraco/Umbraco-CMS/tree/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Web.UI.Client/src/views/documenttypes

mirkomaty commented:
@bjarnef: Since my time currently is very limited, I'm not able to make that code running. That's the reason why I left most of the LeBlender code as it was before. Im aware, that the old code in some areas is very cumbersome, but I didn't have enough time to clean it up. 

If you are able to fix the code and test it, I'd be very pleased to incorporate it into the existing work and prepare a branch for everyone to test and use.

If you are going to fix the "propretyType" typo, please be aware, that the typo is part of the Json code of the configuration files. There should be an automatic conversion of old Json files.

As far as I can see, the legacy adaptor can be removed. It has been around a while and did an automatic conversion of old config files.

Commit:
Cleanup request helper service

Commit:
Fix missing semi colons for named functions

Commit:
Handle catch

Commit:
Remove double check

Commit:
Update dialog title

Commit:
Format document

Commit:
Fix typo in comment

Commit:
Update edit page using umb-editor-header and adjust layout.

Commit:
Update edit page with form inside wrapping container

Commit:
Code cleanup

Commit:
Cleanup output

Commit:
Cleanup unused code

Commit:
Remove legacy adapter

Commit:
Update editorService dialog for LeBlender properties

Commit:
Remove legacy adapter stuff

Commit:
Update close and submit functions

Commit:
Update save and close functions

Commit:
Adjust editor service dialog

Commit:
Format document

Commit:
Cleanup

Commit:
Add vm

Commit:
Save on form submit

Commit:
Update LeBlender editor preview

Commit:
Update remove function

Commit:
Cleanup old stuff

Commit:
Update parameter prevalues

Commit:
Update vm functions

Commit:
Update value

Commit:
Update function name

Commit:
Format document

Commit:
Fix removing function from scope.

Commit:
Update functions

mirkomaty committed:
Merge pull request #1 from bjarnef/leblender-v8-cleanup

LeBlender v8 cleanup

mirkomaty committed:
Bug: New editor wasn't pushed to $scope.editors.

Commit:
Update scope preview function

Commit:
Don't overwrite existing scope model

Commit:
Prevent save grid editor with empty alias or empty name

Commit:
Specify form

Commit:
Make grid editor option required

Commit:
Remove required again to be able to select custom option

Commit:
Cleanup styles

Commit:
Add property sortable options

Commit:
Show expansion panel on multiple items and show confirm when deleting item

Commit:
Always show expand icon in multiple mode

Commit:
Fix position of delete confirm

Commit:
Update naming convension of functions

Commit:
Update add and remove

mirkomaty committed:
Update to Umbraco 8.1

mirkomaty committed:
Umbraco 8.1: Replace AutoMapper with UmbracoMapper.

mirkomaty committed:
camel naming convention

mirkomaty committed:
Merge pull request #2 from bjarnef/leblender-v8-fixes

Fix issue with submit from grid editor dialog

mirkomaty committed:
Avoid including members like $$cache in the property validation.

mirkomaty committed:
Fix the value conversion to use PropertyConverters

mirkomaty committed:
Update the package spec.

mirkomaty committed:
Bug: propertiesOpen value crashes Deserializaton of the LeBlenderModel.

mirkomaty committed:
A better way to exclude the propertiesOpen value from Deserialization.

cleversolutions commented:
Is LeBlender still planning to support v8? It would certainly make migrating V7 content to V8 much easier.

mirkomaty committed:
Quick Hack for a Umbraco Bug with MultiNodeTreePicker.

mirkomaty committed:
Bug: Problems with Umbraco 8.4. Umbraco doesn't save the editor configuration, so we must get the config from the Core.

mirkomaty committed:
Bug: We need to clear the Editor.Config cache, if the config of an editor has been saved.

mirkomaty committed:
Change LeBlender to work with ElementContentTypes.

mirkomaty committed:
Solve problems of wrong redirection after creating and saving a new grid editor

mirkomaty committed:
Update README.md

mirkomaty committed:
Update README.md

mirkomaty committed:
Some editors need an idType parameter in the config in order to use UDIs.
Modified ApplicationInitializer so that the conversion of Grid Editors can be incremental.

mirkomaty committed:
Add error handling in PropertyGridEditorController.

mirkomaty committed:
Bump Assembly Version

mirkomaty committed:
Make sure, that always an Items list exists in LeBlenderModel, even if there are no items in the list.

mirkomaty committed:
Bug after Update to Umbraco 8.7: textarea in Xhtml style can't be rendered correctly so that the LeBlender properties cannot be displayed.

mirkomaty committed:
Update README.md

mirkomaty committed:
Update README.md
